commands/quote.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+const db = require('../services/db');
+
+module.exports = {
+  db_required: true,
+  message_regex: /^.(?:quote)(?: ([^ ]+)|$)(?: (.+)|$)$/,
+  allow: ({ isMod, isAuthenticated, isPM }) => isMod && isAuthenticated && !isPM,
+  response({ message_match: [, command, quote], author_match: [author], channel }) {
+    if (!command) {
+      command = 'get';
+    }
+    switch (command.toLowerCase()) {
+      case 'add':
+      case 'store':
+        return add(author, channel, quote);
+      case 'get':
+        return get(channel, quote);
+      case 'search':
+        return search(channel, quote);
+      case 'help':
+        return [
+          'Add \'quote\' to add a new quote',
+          'Get \'id\' to get a specific quote, or omit id for a random one',
+          'Search \'term\' to search for quote with a searchterm'
+        ];
+      default:
+        return 'Command to store/retrieve quotes. Use `.quote help` for detailed instructions.';
+    }
+  }
+};
+
+function add(author, channel, quote) {
+  if (!quote) {
+    return 'No quote to store.';
+  }
+  const addQuery = 'INSERT INTO `Quote` (`Author`, `Message`, `Chan`) VALUES (?, ?, ?);';
+  return db.conn
+    .query(addQuery, [author, quote, channel])
+    .then(created => `Storing quote '${quote}' from '${author}' on '${channel}', with id '${created.insertId}'`);
```

Comment:
This seems to be the only way you can get the ID of a quote, meaning that if you don't note it right when it was added, it's basically useless to most users. If we're allowing people to get a specific quote by ID, it'd probably be worth returning it in the search results too?

Reply:
Added ID to the search results